saml/src/main/java/com/linecorp/armeria/saml/SamlSessionManager.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.saml;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.CompletionStage;
+
+import javax.annotation.Nullable;
+
+import org.opensaml.messaging.context.MessageContext;
+import org.opensaml.saml.saml2.core.LogoutRequest;
+import org.opensaml.saml.saml2.core.Response;
+
+/**
+ * An interface which is called back when a user is logged in or out.
+ */
+public interface SamlSessionManager {
+
+    /**
+     * A {@link SamlSessionManager} which stores a session into a memory.
```

Comment:
Perhaps we could add something on when this would be used? I guess it's only for testing?

Reply:
Sorry for not enough explanation. I'm trying to figure out how to integrate this SAML module with Central Dogma. So I think I can find how can we use the session manager while the integration is going on. For now, let me move the default session manager to the test code.